ui/pages/smart-transactions/smart-transaction-status-page/smart-transaction-status-page.tsx

```diff
@@ -0,0 +1,314 @@
+  const isSmartTransactionCancelled = smartTransaction?.status?.startsWith(
+    SmartTransactionStatuses.CANCELLED,
+  );
+  const featureFlags: Record<string, any> | null = useSelector(
+    getFeatureFlagsByChainId,
+  );
+  const stxEstimatedDeadline =
+    featureFlags?.smartTransactions?.expectedDeadline || 45;
+  const stxMaxDeadline = featureFlags?.smartTransactions?.maxDeadline || 150;
+  const [timeLeftForPendingStxInSec, setTimeLeftForPendingStxInSec] =
+    useState(stxEstimatedDeadline);
+  const [isSmartTransactionTakingTooLong, setIsSmartTransactionTakingTooLong] =
+    useState(false);
+  const rpcPrefs = useSelector(getRpcPrefsForCurrentProvider, shallowEqual);
+  const chainId: string = useSelector(getCurrentChainId);
+  const stxDeadline = isSmartTransactionTakingTooLong
+    ? stxMaxDeadline
+    : stxEstimatedDeadline;
+
+  const txHash = smartTransaction?.statusMetadata?.minedHash;
+
+  const baseNetworkUrl: string =
+    rpcPrefs.blockExplorerUrl ??
+    SWAPS_CHAINID_DEFAULT_BLOCK_EXPLORER_URL_MAP[chainId] ??
+    '';
+
+  let blockExplorerUrl: string = '';
+  if (txHash) {
+    blockExplorerUrl = getBlockExplorerLink(
+      { hash: txHash, chainId, metamaskNetworkId: '' },
+      { blockExplorerUrl: baseNetworkUrl },
+    );
+  }
+  let title;
+  let description;
+  let iconName;
+  let iconColor;
+  let countdown;
+
+  if (isSmartTransactionPending) {
+    countdown = (
+      <Text
+        display={Display.InlineBlock}
+        textAlign={TextAlign.Center}
+        color={TextColor.textAlternative}
+        variant={TextVariant.bodySm}
+        className="smart-transaction-status-page__countdown"
+      >
+        {showRemainingTimeInMinAndSec(timeLeftForPendingStxInSec)}
+      </Text>
+    );
+  }
+
+  if (isSmartTransactionPending && isSmartTransactionTakingTooLong) {
+    title = t('smartTransactionTakingTooLong');
+    description = t('smartTransactionTakingTooLongDescription', [countdown]);
+    iconName = IconName.Clock;
+    iconColor = IconColor.primaryDefault;
+  } else if (isSmartTransactionPending) {
+    title = t('smartTransactionPending');
+    description = t('stxEstimatedCompletion', [countdown]);
+    iconName = IconName.Clock;
+    iconColor = IconColor.primaryDefault;
+  } else if (isSmartTransactionSuccess) {
+    title = t('smartTransactionSuccess');
+    iconName = IconName.Confirmation;
+    iconColor = IconColor.successDefault;
+  } else if (isSmartTransactionCancelled) {
+    title = t('smartTransactionCancelled');
+    description = t('smartTransactionCancelledDescription');
+    iconName = IconName.Danger;
+    iconColor = IconColor.errorDefault;
+  } else {
+    // E.g. reverted or unknown statuses.
+    title = t('smartTransactionError');
+    description = t('smartTransactionErrorDescription');
+    iconName = IconName.Danger;
+    iconColor = IconColor.errorDefault;
+  }
+
+  useEffect(() => {
+    let intervalId: NodeJS.Timeout;
+    if (isSmartTransactionPending) {
```

Comment:
Rather than nesting the entire block, could we early return here and define the cleanup function prior?